File: release/scripts/smoke-evidence-client-package.js
#!/usr/bin/env node
'use strict';

const assert = require('node:assert');
const { EvidenceClient, EvidenceClientError } = require('@registrystack/evidence-client');

const trustedJwks = {
  keys: [
    {
      kty: 'EC',
      crv: 'P-256',
      alg: 'ES256',
      kid: '_QkPweRjMZxmIHnz7v8tj3coTKx-90L2LRsZbkeP_Bo',
      x: '3kpzAK6fK6xyfqbdp0HvfZCqfgz7MajMviKyM6bsNE4',
      y: 'GkSdSn8xqge52rp9Sv-4qPaw1Q9TJ2eMUyY22flavLU',
    },
  ],
};
const spec = {
  responseFormat: 'signed-jws',
  requirement: 'urn:example:requirement:v1',
  purpose: 'example-purpose',
  audience: 'urn:example:audience',
  evidenceType: 'urn:example:evidence-type:v1',
  issuedBy: 'urn:example:issuer',
  providedBy: 'urn:example:provider',
  configurationRevision: `sha256:${'0'.repeat(64)}`,
  expectedAssuranceProfile: 'local',
  subjects: [{ role: 'subject', selectorProfile: 'national-id' }],
  expectedOutputs: [{ concept: 'urn:example:concept:status-holds', form: 'boolean' }],
  maximumAssertionLifetimeSeconds: 300,
  clockSkewSeconds: 60,
  subjectExpectations: 'acceptFirstUse',
};

// This reserved host and placeholder token make the smoke fail closed if a
// regression unexpectedly attempts network I/O.
const client = new EvidenceClient({
  baseUrl: 'https://evidence.invalid',
  trustedJwks,
  revokedKeyIds: [],
  token: { static: 'placeholder-not-a-credential' },
});
const prepared = client.prepare(spec);
assert.strictEqual(prepared.requestNonce.length, 43);
assert.strictEqual(prepared.policyDocument.audience, spec.audience);
assert.throws(
  () => client.prepare({ ...spec, configurationRevision: '' }),
  (error) => error instanceof EvidenceClientError && error.kind === 'configuration',
);

console.log('Node client package smoke passed');

File: release/scripts/smoke-evidence-client-package.py
#!/usr/bin/env python3
"""Offline smoke for an installed Registry Evidence Python client wheel."""

import registry_evidence_client as client_module


JWKS = {
    "keys": [
        {
            "kty": "EC",
            "crv": "P-256",
            "alg": "ES256",
            "kid": "_QkPweRjMZxmIHnz7v8tj3coTKx-90L2LRsZbkeP_Bo",
            "x": "3kpzAK6fK6xyfqbdp0HvfZCqfgz7MajMviKyM6bsNE4",
            "y": "GkSdSn8xqge52rp9Sv-4qPaw1Q9TJ2eMUyY22flavLU",
        }
    ]
}
SPEC = {
    "response_format": "signed-jws",
    "requirement": "urn:example:requirement:v1",
    "purpose": "example-purpose",
    "audience": "urn:example:audience",
    "evidence_type": "urn:example:evidence-type:v1",
    "issued_by": "urn:example:issuer",
    "provided_by": "urn:example:provider",
    "configuration_revision": "sha256:" + "0" * 64,
    "expected_assurance_profile": "local",
    "subjects": [{"role": "subject", "selector_profile": "national-id"}],
    "expected_outputs": [
        {"concept": "urn:example:concept:status-holds", "form": "boolean"}
    ],
    "maximum_assertion_lifetime_seconds": 300,
    "clock_skew_seconds": 60,
    "subject_expectations": "accept_first_use",
}


def main() -> None:
    # This reserved host and placeholder token make the smoke fail closed if a
    # regression unexpectedly attempts network I/O.
    client = client_module.EvidenceClient(
        "https://evidence.invalid", JWKS, [], "placeholder-not-a-credential"
    )
    prepared = client.prepare(SPEC)
    if len(prepared.request_nonce) != 43:
        raise SystemExit(f"unexpected nonce length {len(prepared.request_nonce)}")
    if prepared.policy_document["audience"] != SPEC["audience"]:
        raise SystemExit("the prepared policy does not carry the requested audience")
    if prepared.subject_expectations != "accept_first_use":
        raise SystemExit("the prepared request lost its subject expectations")

    try:
        client.prepare(dict(SPEC, configuration_revision=""))
    except client_module.ConfigurationError as error:
        if error.kind != "configuration":
            raise SystemExit(f"unexpected error kind {error.kind!r}") from error
    else:
        raise SystemExit("an empty configuration revision must be refused")

    print("Python client package smoke passed")


if __name__ == "__main__":
    main()
